hyperspy/misc/label_position.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2020 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import copy
+import itertools
+
+import numpy as np
+from hyperspy.misc.eels.tools import get_edges_near_energy, get_info_from_edges
```

Comment:
This file should be not be signal specific and generic enough to support different type of signal: would it be possible to `get_info_from_edges`?

Reply:
Removed all dependency on `hyperspy.misc.eels.tools`